.github/workflows/CI_main.yml

```diff
@@ -0,0 +1,47 @@
+name: Tostadas CI
+
+on:
+  pull_request:
+    types: [opened, reopened]
+    branches: [master]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout Code
+      uses: actions/checkout@v2
+      with:
+          ref: ${{ github.event.pull_request.head.ref }}
+
+    - name: Set Up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.9'
+
+    - name: Install Miniconda
+      run: |
+        curl -s -LO https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh
+        bash Miniconda3-latest-Linux-x86_64.sh -b -p $HOME/miniconda
+        export PATH="$HOME/miniconda/bin:$PATH"
+
+    - name: Verify Conda installation
+      run: conda --version
+
+    - name: Initialize Conda shell
+      run: |
+        echo ". $HOME/miniconda/etc/profile.d/conda.sh" >> $HOME/.bashrc
+        echo "conda activate base" >> $HOME/.bashrc
+
+    - name: Create Environment and Run Tests 
+      run: |
+        source ~/miniconda/etc/profile.d/conda.sh
+        conda install mamba --quiet -n base -c conda-forge
+        mamba env create --quiet -n ci_tostadas -f environment.yml
+        conda activate ci_tostadas
```

Comment:
Here and below, why still use conda? I would use mamba as soon as possible early on and never use conda thereafter if possible. Can also install mamba like this and avoid conda  all together. 

# Install Mambaforge
curl -L -O https://github.com/conda-forge/miniforge/releases/latest/download/Mambaforge-$(uname)-$(uname -m).sh
bash Mambaforge-$(uname)-$(uname -m).sh -b -p $HOME/mambaforge

# Move conda/mamba to path
export PATH="$HOME/mambaforge/bin:$PATH

Reply:
Made changes to the .yaml file to use Mambaforge and mamba for all package installs